Fix movement and rotation

Faffie opened the pull request:
Fixes movement and rotation.
Apply deadzone for Gamepads.
Few minor corrections.

Faffie committed:
Fix movement/rotation

irenevanderblij commented:
Not sure where this went wrong, but great that you fixed it!!
Everything seems to work.
The inversion of the looking up and down arrows is kind of weird to me, I don't see how this makes more sense. But since players do not use this I will not make a big issue out of this.
Would be great if this was merged quickly.

alanvanrossum commented:
How did this even happen in the first place? Any clue?

Faffie commented:
The code from yesterday was rotating along the wrong axis. To rotate left, you should rotate along Y, not X etc.
Input command strings differed ("tilt" vs "Tilt") and thus weren't recognised.

Then, my mistakes:
I made a typo writing "winning" (didn't affect gameplay). 
When I implemented the lock on the Y-axis, setting it to 0 worked fine (spawn position y was set to 0 as well). Later on, I let the code get the Y component from the spawn vector. Because this was set to 0 as well, everything still worked fine. Because I changed the spawn position later on to 0,2,0 and because the code for the lock would use the same y component, movement was always set upwards. Movement vectors are relative, not absolute, duh. :$ I didn't realise what I had done until we noticed this bug. Sorry. :(
